tests/PHPStan/Rules/PhpDoc/IncompatiblePhpDocTypeRuleTest.php

```diff
@@ -233,4 +233,23 @@ public function testConditionalReturnType(): void
 		]);
 	}
 
+	public function testParamOut(): void
+	{
+		$this->analyse([__DIR__ . '/data/param-out.php'], [
+			[
+				'PHPDoc tag @param-out references unknown parameter: $z',
+				23,
+			],
+			[
+				'PHPDoc tag @param-out for parameter $i with type string is incompatible with native type int.',
```

Comment:
Why does the param-out type need to be compatible?

Reply:
good catch. by-ref seems to be allowed to overwrite indepent from the parameters type
https://3v4l.org/W9C3X

fixed in the latest push